### PR TITLE
remove extraneous io exceptions + improve generic infrastructure for log...

### PR DESCRIPTION
**Pull request by dwightguth**
...ging errors

@cos please review

### CODE REVIEW

File: src/javasources/KTool/src/org/kframework/backend/kore/KoreBackend.java [hunk 1]
```diff
@@ -74,10 +75,15 @@ public void run(Definition toKore) throws IOException {
 
           if(!fileTable.containsKey(((toKore.getItems().get(i)).getFilename()))){
 
+              String filename = ((toKore.getItems().get(i)).getFilename().substring(0,
+                      (toKore.getItems().get(i)).getFilename().length()-2))+".kore";
```
Comment (cos) [inline src/javasources/KTool/src/org/kframework/backend/kore/KoreBackend.java:79]: minor: extract `toKore.getItems().get(i)).getFilename()` into a variable

**Comment by cos** [comment]
This looks good. It gets us stack traces in most cases we want them. Not sure about this, but it may still be good for `KException` to piggyback the original exception. It could be useful when, for example, we're debugging without the verbose option and end up with a few exceptions in `kem`. The exceptions could be inspected for their original source. Would that be hard?

File: src/javasources/KTool/src/org/kframework/backend/kore/KoreBackend.java (hunk 1 above)
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/backend/kore/KoreBackend.java:79]: I'd rather not... this is code that is on a deprecated code path and I don't really particularly want to mess with it any more than I have to...

**Comment by dwightguth** [comment]
@cos please re-review

**Comment by dwightguth** [comment]
Cosmin approved this via IM to me so I am merging this.

**Pull request merged**